ui/src/app/lib/atlasmap-data-mapper/components/document.definition.component.ts

```diff
@@ -197,8 +197,8 @@ export class DocumentDefinitionComponent {
         const self: DocumentDefinitionComponent = this;
         this.modalWindow.reset();
         this.modalWindow.confirmButtonText = 'Save';
-        const isProperty: boolean = docDef.initCfg.type.isProperty();
-        const isConstant: boolean = docDef.initCfg.type.isConstant();
+        const isProperty: boolean = docDef.type == DocumentType.PROPERTY;
+        const isConstant: boolean = docDef.type == DocumentType.CONSTANT;
```

Comment:
This is just a quick note regarding styling, but nothing really mandatory at all:

TypeScript is smart enough to infer types out of the assigned expression of a class field. In such case as the above, you can safely remove the `boolean` type annotation from the const expression. 

Feel free to remove it here or leave it, and bear in mind this hint for the future. ;)

Reply:
Oh good to know. This is a bad habit Java has been enforcing on me :-)